CRITICAL - 2020-06-19 14:23:37 --> No connection could be made because the target machine actively refused it.

#0 C:\xampp\htdocs\rpg\vendor\codeigniter4\framework\system\Database\BaseConnection.php(377): CodeIgniter\Database\MySQLi\Connection->connect(false)
#1 C:\xampp\htdocs\rpg\vendor\codeigniter4\framework\system\Database\BaseConnection.php(643): CodeIgniter\Database\BaseConnection->initialize()
#2 C:\xampp\htdocs\rpg\vendor\codeigniter4\framework\system\Database\BaseBuilder.php(1845): CodeIgniter\Database\BaseConnection->query('SELECT *\nFROM `...', Array, false)
#3 C:\xampp\htdocs\rpg\vendor\codeigniter4\framework\system\Model.php(489): CodeIgniter\Database\BaseBuilder->get()
#4 C:\xampp\htdocs\rpg\app\Validation\UserRules.php(12): CodeIgniter\Model->first()
#5 C:\xampp\htdocs\rpg\vendor\codeigniter4\framework\system\Validation\Validation.php(294): App\Validation\UserRules->validateUser('andikarizki', 'username,passwo...', Array, NULL)
#6 C:\xampp\htdocs\rpg\vendor\codeigniter4\framework\system\Validation\Validation.php(189): CodeIgniter\Validation\Validation->processRules('password', 'password', 'andikarizki', Array, Array)
#7 C:\xampp\htdocs\rpg\vendor\codeigniter4\framework\system\Controller.php(219): CodeIgniter\Validation\Validation->run()
#8 C:\xampp\htdocs\rpg\app\Controllers\Auth.php(28): CodeIgniter\Controller->validate(Array, Array)
#9 C:\xampp\htdocs\rpg\vendor\codeigniter4\framework\system\CodeIgniter.php(914): App\Controllers\Auth->index()
#10 C:\xampp\htdocs\rpg\vendor\codeigniter4\framework\system\CodeIgniter.php(400): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Auth))
#11 C:\xampp\htdocs\rpg\vendor\codeigniter4\framework\system\CodeIgniter.php(308): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#12 C:\xampp\htdocs\rpg\public\index.php(45): CodeIgniter\CodeIgniter->run()
#13 C:\xampp\htdocs\rpg\vendor\codeigniter4\framework\system\Commands\Server\rewrite.php(34): require_once('C:\\xampp\\htdocs...')
#14 {main}
CRITICAL - 2020-06-19 14:47:50 --> No connection could be made because the target machine actively refused it.

#0 C:\xampp\htdocs\rpg\vendor\codeigniter4\framework\system\Database\BaseConnection.php(377): CodeIgniter\Database\MySQLi\Connection->connect(false)
#1 C:\xampp\htdocs\rpg\vendor\codeigniter4\framework\system\Database\BaseConnection.php(643): CodeIgniter\Database\BaseConnection->initialize()
#2 C:\xampp\htdocs\rpg\vendor\codeigniter4\framework\system\Database\BaseBuilder.php(1845): CodeIgniter\Database\BaseConnection->query('SELECT *\nFROM `...', Array, false)
#3 C:\xampp\htdocs\rpg\vendor\codeigniter4\framework\system\Model.php(489): CodeIgniter\Database\BaseBuilder->get()
#4 C:\xampp\htdocs\rpg\app\Validation\UserRules.php(12): CodeIgniter\Model->first()
#5 C:\xampp\htdocs\rpg\vendor\codeigniter4\framework\system\Validation\Validation.php(294): App\Validation\UserRules->validateUser('andikarizki', 'username,passwo...', Array, NULL)
#6 C:\xampp\htdocs\rpg\vendor\codeigniter4\framework\system\Validation\Validation.php(189): CodeIgniter\Validation\Validation->processRules('password', 'password', 'andikarizki', Array, Array)
#7 C:\xampp\htdocs\rpg\vendor\codeigniter4\framework\system\Controller.php(219): CodeIgniter\Validation\Validation->run()
#8 C:\xampp\htdocs\rpg\app\Controllers\Auth.php(28): CodeIgniter\Controller->validate(Array, Array)
#9 C:\xampp\htdocs\rpg\vendor\codeigniter4\framework\system\CodeIgniter.php(914): App\Controllers\Auth->index()
#10 C:\xampp\htdocs\rpg\vendor\codeigniter4\framework\system\CodeIgniter.php(400): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Auth))
#11 C:\xampp\htdocs\rpg\vendor\codeigniter4\framework\system\CodeIgniter.php(308): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#12 C:\xampp\htdocs\rpg\public\index.php(45): CodeIgniter\CodeIgniter->run()
#13 C:\xampp\htdocs\rpg\vendor\codeigniter4\framework\system\Commands\Server\rewrite.php(34): require_once('C:\\xampp\\htdocs...')
#14 {main}
CRITICAL - 2020-06-19 14:54:18 --> Undefined variable: playerDamageFormula
#0 C:\xampp\htdocs\rpg\app\Controllers\MatchController.php(58): CodeIgniter\Debug\Exceptions->errorHandler(8, 'Undefined varia...', 'C:\\xampp\\htdocs...', 58, Array)
#1 C:\xampp\htdocs\rpg\vendor\codeigniter4\framework\system\CodeIgniter.php(914): App\Controllers\MatchController->MatchSimulation()
#2 C:\xampp\htdocs\rpg\vendor\codeigniter4\framework\system\CodeIgniter.php(400): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\MatchController))
#3 C:\xampp\htdocs\rpg\vendor\codeigniter4\framework\system\CodeIgniter.php(308): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#4 C:\xampp\htdocs\rpg\public\index.php(45): CodeIgniter\CodeIgniter->run()
#5 C:\xampp\htdocs\rpg\vendor\codeigniter4\framework\system\Commands\Server\rewrite.php(34): require_once('C:\\xampp\\htdocs...')
#6 {main}
CRITICAL - 2020-06-19 14:56:54 --> Undefined variable: playerDamageFormula
#0 C:\xampp\htdocs\rpg\app\Controllers\MatchController.php(58): CodeIgniter\Debug\Exceptions->errorHandler(8, 'Undefined varia...', 'C:\\xampp\\htdocs...', 58, Array)
#1 C:\xampp\htdocs\rpg\vendor\codeigniter4\framework\system\CodeIgniter.php(914): App\Controllers\MatchController->MatchSimulation()
#2 C:\xampp\htdocs\rpg\vendor\codeigniter4\framework\system\CodeIgniter.php(400): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\MatchController))
#3 C:\xampp\htdocs\rpg\vendor\codeigniter4\framework\system\CodeIgniter.php(308): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#4 C:\xampp\htdocs\rpg\public\index.php(45): CodeIgniter\CodeIgniter->run()
#5 C:\xampp\htdocs\rpg\vendor\codeigniter4\framework\system\Commands\Server\rewrite.php(34): require_once('C:\\xampp\\htdocs...')
#6 {main}
CRITICAL - 2020-06-19 17:34:11 --> Undefined variable: mobCurrentHP
#0 C:\xampp\htdocs\rpg\app\Views\match.php(32): CodeIgniter\Debug\Exceptions->errorHandler(8, 'Undefined varia...', 'C:\\xampp\\htdocs...', 32, Array)
#1 C:\xampp\htdocs\rpg\vendor\codeigniter4\framework\system\View\View.php(246): include('C:\\xampp\\htdocs...')
#2 C:\xampp\htdocs\rpg\vendor\codeigniter4\framework\system\Common.php(1087): CodeIgniter\View\View->render('/match', Array, true)
#3 C:\xampp\htdocs\rpg\app\Controllers\MatchController.php(63): view('/match', Array)
#4 C:\xampp\htdocs\rpg\vendor\codeigniter4\framework\system\CodeIgniter.php(914): App\Controllers\MatchController->MatchSimulation()
#5 C:\xampp\htdocs\rpg\vendor\codeigniter4\framework\system\CodeIgniter.php(400): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\MatchController))
#6 C:\xampp\htdocs\rpg\vendor\codeigniter4\framework\system\CodeIgniter.php(308): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#7 C:\xampp\htdocs\rpg\public\index.php(45): CodeIgniter\CodeIgniter->run()
#8 C:\xampp\htdocs\rpg\vendor\codeigniter4\framework\system\Commands\Server\rewrite.php(34): require_once('C:\\xampp\\htdocs...')
#9 {main}
CRITICAL - 2020-06-19 17:36:03 --> Cannot use a scalar value as an array
#0 C:\xampp\htdocs\rpg\app\Controllers\MatchController.php(97): CodeIgniter\Debug\Exceptions->errorHandler(2, 'Cannot use a sc...', 'C:\\xampp\\htdocs...', 97, Array)
#1 C:\xampp\htdocs\rpg\app\Controllers\MatchController.php(54): App\Controllers\MatchController->MatchAttack(-5)
#2 C:\xampp\htdocs\rpg\vendor\codeigniter4\framework\system\CodeIgniter.php(914): App\Controllers\MatchController->MatchSimulation()
#3 C:\xampp\htdocs\rpg\vendor\codeigniter4\framework\system\CodeIgniter.php(400): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\MatchController))
#4 C:\xampp\htdocs\rpg\vendor\codeigniter4\framework\system\CodeIgniter.php(308): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#5 C:\xampp\htdocs\rpg\public\index.php(45): CodeIgniter\CodeIgniter->run()
#6 C:\xampp\htdocs\rpg\vendor\codeigniter4\framework\system\Commands\Server\rewrite.php(34): require_once('C:\\xampp\\htdocs...')
#7 {main}
CRITICAL - 2020-06-19 17:36:30 --> Cannot use a scalar value as an array
#0 C:\xampp\htdocs\rpg\app\Controllers\MatchController.php(97): CodeIgniter\Debug\Exceptions->errorHandler(2, 'Cannot use a sc...', 'C:\\xampp\\htdocs...', 97, Array)
#1 C:\xampp\htdocs\rpg\app\Controllers\MatchController.php(54): App\Controllers\MatchController->MatchAttack(-5)
#2 C:\xampp\htdocs\rpg\vendor\codeigniter4\framework\system\CodeIgniter.php(914): App\Controllers\MatchController->MatchSimulation()
#3 C:\xampp\htdocs\rpg\vendor\codeigniter4\framework\system\CodeIgniter.php(400): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\MatchController))
#4 C:\xampp\htdocs\rpg\vendor\codeigniter4\framework\system\CodeIgniter.php(308): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#5 C:\xampp\htdocs\rpg\public\index.php(45): CodeIgniter\CodeIgniter->run()
#6 C:\xampp\htdocs\rpg\vendor\codeigniter4\framework\system\Commands\Server\rewrite.php(34): require_once('C:\\xampp\\htdocs...')
#7 {main}
CRITICAL - 2020-06-19 17:36:53 --> Cannot use a scalar value as an array
#0 C:\xampp\htdocs\rpg\app\Controllers\MatchController.php(97): CodeIgniter\Debug\Exceptions->errorHandler(2, 'Cannot use a sc...', 'C:\\xampp\\htdocs...', 97, Array)
#1 C:\xampp\htdocs\rpg\app\Controllers\MatchController.php(54): App\Controllers\MatchController->MatchAttack(-5)
#2 C:\xampp\htdocs\rpg\vendor\codeigniter4\framework\system\CodeIgniter.php(914): App\Controllers\MatchController->MatchSimulation()
#3 C:\xampp\htdocs\rpg\vendor\codeigniter4\framework\system\CodeIgniter.php(400): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\MatchController))
#4 C:\xampp\htdocs\rpg\vendor\codeigniter4\framework\system\CodeIgniter.php(308): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#5 C:\xampp\htdocs\rpg\public\index.php(45): CodeIgniter\CodeIgniter->run()
#6 C:\xampp\htdocs\rpg\vendor\codeigniter4\framework\system\Commands\Server\rewrite.php(34): require_once('C:\\xampp\\htdocs...')
#7 {main}
CRITICAL - 2020-06-19 17:42:08 --> Cannot use a scalar value as an array
#0 C:\xampp\htdocs\rpg\app\Controllers\MatchController.php(97): CodeIgniter\Debug\Exceptions->errorHandler(2, 'Cannot use a sc...', 'C:\\xampp\\htdocs...', 97, Array)
#1 C:\xampp\htdocs\rpg\app\Controllers\MatchController.php(54): App\Controllers\MatchController->MatchAttack(-5)
#2 C:\xampp\htdocs\rpg\vendor\codeigniter4\framework\system\CodeIgniter.php(914): App\Controllers\MatchController->MatchSimulation()
#3 C:\xampp\htdocs\rpg\vendor\codeigniter4\framework\system\CodeIgniter.php(400): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\MatchController))
#4 C:\xampp\htdocs\rpg\vendor\codeigniter4\framework\system\CodeIgniter.php(308): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#5 C:\xampp\htdocs\rpg\public\index.php(45): CodeIgniter\CodeIgniter->run()
#6 C:\xampp\htdocs\rpg\vendor\codeigniter4\framework\system\Commands\Server\rewrite.php(34): require_once('C:\\xampp\\htdocs...')
#7 {main}
CRITICAL - 2020-06-19 17:42:42 --> A non-numeric value encountered
#0 C:\xampp\htdocs\rpg\app\Controllers\MatchController.php(102): CodeIgniter\Debug\Exceptions->errorHandler(2, 'A non-numeric v...', 'C:\\xampp\\htdocs...', 102, Array)
#1 C:\xampp\htdocs\rpg\app\Controllers\MatchController.php(54): App\Controllers\MatchController->MatchAttack(-5)
#2 C:\xampp\htdocs\rpg\vendor\codeigniter4\framework\system\CodeIgniter.php(914): App\Controllers\MatchController->MatchSimulation()
#3 C:\xampp\htdocs\rpg\vendor\codeigniter4\framework\system\CodeIgniter.php(400): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\MatchController))
#4 C:\xampp\htdocs\rpg\vendor\codeigniter4\framework\system\CodeIgniter.php(308): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#5 C:\xampp\htdocs\rpg\public\index.php(45): CodeIgniter\CodeIgniter->run()
#6 C:\xampp\htdocs\rpg\vendor\codeigniter4\framework\system\Commands\Server\rewrite.php(34): require_once('C:\\xampp\\htdocs...')
#7 {main}
CRITICAL - 2020-06-19 17:43:02 --> A non-numeric value encountered
#0 C:\xampp\htdocs\rpg\app\Controllers\MatchController.php(102): CodeIgniter\Debug\Exceptions->errorHandler(2, 'A non-numeric v...', 'C:\\xampp\\htdocs...', 102, Array)
#1 C:\xampp\htdocs\rpg\app\Controllers\MatchController.php(54): App\Controllers\MatchController->MatchAttack(-5)
#2 C:\xampp\htdocs\rpg\vendor\codeigniter4\framework\system\CodeIgniter.php(914): App\Controllers\MatchController->MatchSimulation()
#3 C:\xampp\htdocs\rpg\vendor\codeigniter4\framework\system\CodeIgniter.php(400): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\MatchController))
#4 C:\xampp\htdocs\rpg\vendor\codeigniter4\framework\system\CodeIgniter.php(308): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#5 C:\xampp\htdocs\rpg\public\index.php(45): CodeIgniter\CodeIgniter->run()
#6 C:\xampp\htdocs\rpg\vendor\codeigniter4\framework\system\Commands\Server\rewrite.php(34): require_once('C:\\xampp\\htdocs...')
#7 {main}
CRITICAL - 2020-06-19 17:43:20 --> A non-numeric value encountered
#0 C:\xampp\htdocs\rpg\app\Controllers\MatchController.php(102): CodeIgniter\Debug\Exceptions->errorHandler(2, 'A non-numeric v...', 'C:\\xampp\\htdocs...', 102, Array)
#1 C:\xampp\htdocs\rpg\app\Controllers\MatchController.php(54): App\Controllers\MatchController->MatchAttack(-5)
#2 C:\xampp\htdocs\rpg\vendor\codeigniter4\framework\system\CodeIgniter.php(914): App\Controllers\MatchController->MatchSimulation()
#3 C:\xampp\htdocs\rpg\vendor\codeigniter4\framework\system\CodeIgniter.php(400): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\MatchController))
#4 C:\xampp\htdocs\rpg\vendor\codeigniter4\framework\system\CodeIgniter.php(308): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#5 C:\xampp\htdocs\rpg\public\index.php(45): CodeIgniter\CodeIgniter->run()
#6 C:\xampp\htdocs\rpg\vendor\codeigniter4\framework\system\Commands\Server\rewrite.php(34): require_once('C:\\xampp\\htdocs...')
#7 {main}
CRITICAL - 2020-06-19 17:43:27 --> A non-numeric value encountered
#0 C:\xampp\htdocs\rpg\app\Controllers\MatchController.php(102): CodeIgniter\Debug\Exceptions->errorHandler(2, 'A non-numeric v...', 'C:\\xampp\\htdocs...', 102, Array)
#1 C:\xampp\htdocs\rpg\app\Controllers\MatchController.php(54): App\Controllers\MatchController->MatchAttack(-5)
#2 C:\xampp\htdocs\rpg\vendor\codeigniter4\framework\system\CodeIgniter.php(914): App\Controllers\MatchController->MatchSimulation()
#3 C:\xampp\htdocs\rpg\vendor\codeigniter4\framework\system\CodeIgniter.php(400): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\MatchController))
#4 C:\xampp\htdocs\rpg\vendor\codeigniter4\framework\system\CodeIgniter.php(308): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#5 C:\xampp\htdocs\rpg\public\index.php(45): CodeIgniter\CodeIgniter->run()
#6 C:\xampp\htdocs\rpg\vendor\codeigniter4\framework\system\Commands\Server\rewrite.php(34): require_once('C:\\xampp\\htdocs...')
#7 {main}
CRITICAL - 2020-06-19 17:48:54 --> Cannot use [] for reading
#0 [internal function]: CodeIgniter\Debug\Exceptions->shutdownHandler()
#1 {main}
